modules/rest_api/controllers/services/rest.php

```diff
@@ -1731,6 +2126,201 @@ private function esGetSpecialFieldHigherGeography(array $doc, array $params) {
     }
   }
 
+  /**
+   * Special field handler for Elasticsearch location VC with format options.
+   *
+   * Converts Vice County code to values as specified in format option.
+   * 
+   * @param array $doc
+   *   Elasticsearch document.
+   * @param array $params
+   *   An identifier for the format.
+   *
+   * @return string
+   *   Formatted VC.
+   */
+  private function esGetSpecialFieldVc(array $doc, array $params) {
```

Comment:
I'm aware that VC is a concept specific to the UK and also that the '#higher_geography:Vice County:code#' format nearly does what is required here, apart from the swapping of empty values for zeros. Can this be handled by a format specifier instead of a separate VC field?

Reply:
I will revisit.

Reply:
I've done this - I agree it's better than a dedicated VC formatter. The HigherGeography handler already accommodated a third parameter to format output (which could be set to 'json') so I have allowed this to be set to 'mapmate'. Note that it isn't just empty values that need to be replaced with zeros for MapMate, but also instances where more than one VC number is returned. It is specified like this: `#higher_geography:Vice County:code:mapmate#`. It's a little clunky in that there's nothing to stop it being used like this: `#higher_geography:Vice County:name:mapmate#` which would output either a single VC name or a zero.